core/src/main/java/com/datastax/oss/driver/api/core/NodeUnavailableException.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * Indicates that a Node encountered as part of processing a query plan has no connections
+ * available.
+ *
+ * <p>This might happen under heavy load. The driver will automatically try the next node in the
+ * query plan. Therefore, the only way that the client can observe this exception is as part of a
+ * {@link AllNodesFailedException}.
+ * <p>The default load balancer will attempt to evenly distribute requests across nodes, so a common
```

Comment:
That's a good explanation, but I wouldn't mention "load balancer" and "evenly distribute" here. I would rephrase this paragraph and the following one as follows:

```
 * <p>A common reason to encounter this error is when the configured number of connections per node
 * and requests per connection is not high enough to absorb the overall request rate. This can be
 * mitigated by tuning the following options:
 *
 * <ul>
 *   <li>{@code advanced.connection.pool.local.size};
 *   <li>{@code advanced.connection.pool.remote.size};
 *   <li>{@code advanced.connection.max-requests-per-connection}.
 * </ul>
 *
 * See {@code reference.conf} for more details.
 *
 * <p>Another possibility is when you are trying to direct a request {@linkplain
 * com.datastax.oss.driver.api.core.cql.Statement#setNode(Node) to a particular node}, but that node
 * has no connections available.
```

Reply:
done and done